### DISCUSSION TITLE
Game console support?

### DISCUSSION THREAD

**Discussion by rubenlg** [vsg-dev/VulkanSceneGraph#885] [Q&A]
I've been using OSG on and off since 2000 and am now faced with the decision to continue using what I know well or switch to VSG. Considering OSG is in maintenance mode, VSG seems a bit more attractive. My use case doesn't involve cutting-edge graphics, so I'm fine with either of them.

I plan to have game console support for my code in the near future, and while I'm fairly confident that OSG+SDL will work without too much effort (I know SDL is supported while OSG has very little platform-specific code besides viewers and works even on old compilers), I'm not so confident about VSG.

I'm aware that console SDKs have strict NDAs and that it's hard to talk about their tech details on a public forum, but (perhaps avoiding specifics) has anyone had success using it on modern consoles without going through a complex port of the VSG library?

**Comment by robertosfield** [comment]
I haven't had any discussions about console support for the VSG. External dependencies of the VSG itself is just C++17, Vulkan and CMake + native Windowing so as long as they are available it should be easy to port across. 

I don't know if SDL1,2 or 3 would make any difference with how easy it would be to create console support but you can compile the VSG without windowing and then just use SDL.  There is a vsgSDL library you could look at for reference:

https://github.com/sum1afk/vsgSDL

**Comment by rubenlg** [comment]
C++17 might be the most challenging part. I don't have access to the SDKs yet, but from the little info I could gather from the Internet, it seems that some console SDKs tend to ship with oldish compilers. Hopefully that will become clearer once I manage to get access to the SDKs. If you don't mind, I'll keep this question open a little longer in case anyone from the community had some experience with VSG in consoles.

Thanks for the vsgSDL pointer!
